src/app-layer-ftp.c

```diff
@@ -470,9 +550,31 @@ static int FTPParseRequest(Flow *f, void *ftp_state,
 
     int direction = STREAM_TOSERVER;
     while (FTPGetLine(state) >= 0) {
-        FTPParseRequestCommand(state,
-                               state->current_line, state->current_line_len);
+        const FtpCommand *cmd_descriptor;
+
+        if (!FTPParseRequestCommand(state->current_line, state->current_line_len, &cmd_descriptor)) {
```

Comment:
this check means that if we encounter some custom or unsupported command we won't create a tx, right?

Reply:
Correct; if the command isn't parsed there won't be a transaction.